src/utils/opResult/index.ts

```diff
@@ -1,10 +1,13 @@
-export type OpResult<T> = Promise<
```

Comment:
Making `OpResult` into a regular result results in this being not just promise specific, plus it works more nicely in TS because you can't `await OpResult<Promise>` but you can `await Promise<OpResult>`